glue/normaliser.py

```
@@ -0,0 +1,221 @@
+            "run_id": essentials.get("firedDateTime", ""),
+            "failed_task": target,
+            "error_message": error_msg,
+            "alert_rule": rule,
+            **target_meta,
+        },
+    }
+
+
+def make_alert_dedup_key(job_id: str | int, error_type: str) -> str:
+    raw = f"{job_id}:{error_type}"
+    return hashlib.md5(raw.encode()).hexdigest()
```

Ruff (0.15.9)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)